### PR TITLE
Test specific generators for kubectl

### PR DESCRIPTION
**Pull request by liggitt**
As we add more, we'll want to keep these to ensure previous generators continue working in the same way.

### CODE REVIEW

**Comment by ixdy** [comment]
Can someone in @kubernetes/goog-csi take this review over?

**Comment by ixdy** [comment]
@liggitt e2e failure looks related to this PR:
```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:863
Expected error:
    <*errors.errorString | 0xc8204681b0>: {
        s: "Error running &{/var/lib/jenkins/workspace/kubernetes-pull-build-test-e2e-gce@2/kubernetes/platforms/linux/amd64/kubectl [kubectl --server=https://104.154.27.45 --kubeconfig=/var/lib/jenkins/workspace/kubernetes-pull-build-test-e2e-gce@2/.kube/config run e2e-test-nginx-rc --generator=run-controller/v1 --image=nginx --namespace=e2e-tests-kubectl-sb8ru] []  <nil>  error: generator \"run-controller/v1\" not found.\nSee 'kubectl run -h' for help and examples.\n [] <nil> 0xc820a4f160 exit status 1 <nil> true [0xc820058828 0xc820058840 0xc820058858] [0xc820058828 0xc820058840 0xc820058858] [0xc820058838 0xc820058850] [0x88d060 0x88d060] 0xc820a4aae0}:\nCommand stdout:\n\nstderr:\nerror: generator \"run-controller/v1\" not found.\nSee 'kubectl run -h' for help and examples.\n\nerror:\nexit status 1\n",
    }
    Error running &{/var/lib/jenkins/workspace/kubernetes-pull-build-test-e2e-gce@2/kubernetes/platforms/linux/amd64/kubectl [kubectl --server=https://104.154.27.45 --kubeconfig=/var/lib/jenkins/workspace/kubernetes-pull-build-test-e2e-gce@2/.kube/config run e2e-test-nginx-rc --generator=run-controller/v1 --image=nginx --namespace=e2e-tests-kubectl-sb8ru] []  <nil>  error: generator "run-controller/v1" not found.
    See 'kubectl run -h' for help and examples.
     [] <nil> 0xc820a4f160 exit status 1 <nil> true [0xc820058828 0xc820058840 0xc820058858] [0xc820058828 0xc820058840 0xc820058858] [0xc820058838 0xc820058850] [0x88d060 0x88d060] 0xc820a4aae0}:
    Command stdout:
    
    stderr:
    error: generator "run-controller/v1" not found.
    See 'kubectl run -h' for help and examples.
    
    error:
    exit status 1
    
not to have occurred
```

**Comment by liggitt** [comment]
Will look

File: test/e2e/kubectl.go [hunk 1]
```diff
@@ -833,7 +833,7 @@ var _ = Describe("Kubectl client", func() {
 			image := "nginx"
 
 			By("running the image " + image)
-			runKubectlOrDie("run", rcName, "--image="+image, nsFlag)
+			runKubectlOrDie("run", rcName, "--generator=run-controller/v1", "--image="+image, nsFlag)
```
Comment (kargakis) [inline test/e2e/kubectl.go]: We use this in Origin as run-controller/v1 and reserve run/v1 for running DCs. Should be [run/v1](https://github.com/kubernetes/kubernetes/blob/06996d5af95a8c283f670ffebd09adf9396d82dd/pkg/kubectl/cmd/util/factory.go#L119).

Comment (liggitt) [inline test/e2e/kubectl.go]: Yeah, I'm rethinking how I want to do this

**Comment by lavalamp** [comment]
This looks like a good idea to me.

**Pull request closed without merging**